pkg/lock/lock_debug.go

```diff
@@ -81,6 +91,12 @@ func (i *internalMutex) Lock() {
 	i.Time = time.Now()
 }
 
+func (i *internalMutex) TryLock() bool {
+	res := i.Mutex.TryLock()
+	i.Time = time.Now()
```

Comment:
should this write to `i.Time` if res is `false`?

Reply:
Maybe not, but it seems also harmless, as `i.Time` is only used in `Unlock` and in the case `TryLock` failing there is always another write to `i.Time` before the next `Unlock`.

Made the setting of `i.Time` condtional on `res`.

Reply:
Ah, yeah that reasoning makes a lot of sense- either way, resolved.